ygen/directory.go

```diff
@@ -89,19 +89,17 @@ func GetOrderedFieldNames(directory *Directory) []string {
 }
 
 // GoFieldNameMap returns a map containing the Go name for a field (key
-// is the field schema name). Camelcase and uniquification is done to ensure
-// compilation. Naming uniquification is done deterministically.
-func GoFieldNameMap(directory *Directory) map[string]string {
+// is the field schema name). Name Uniquification is done to ensure
```

Comment:
Perhaps I'm missing something here - `GoFieldNameMap` seems like it should return a map of a field to a Go field name -- if you don't create the camel case name here, then isn't this actually not creating a Go field name?

Reply:
It's now operating on `ParsedDirectory`, which has the candidate Go names already populated via GoLangMapper.
Here: `directory.Fields[fName].Name`

Reply:
So here, what is the value of this map? All the examples seem to have the same values corresponding to one another. Can we eliminate it?

Reply:
`ParsedDirectory.Fields` doc comment explains why a map is needed here. Does this answer your question?
https://github.com/openconfig/ygot/blob/6f722a0cce2a47949294afa0c3f23b080d51e501/ygen/ir.go#L209-L212

Reply:
Ah, I see, thanks -- missed that there might be collisions here.